src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -131,10 +133,70 @@ std::shared_ptr<uint32_t> GetCounterHolder(const char * name)
 namespace chip {
 namespace Dnssd {
 
+/**
+ * @brief Returns the domain name from a given hostname with domain.
+ *        The assumption here is that the hostname comprises of "hostnameWithoutDomain.<domain>."
+ *        The domainName returned from this API is "<domain>."
+ *
+ * @param[in] hostname The hostname with domain.
+ */
+std::string GetDomainFromHostName(const char * hostnameWithDomain)
+{
+    std::string hostname = std::string(hostnameWithDomain);
+
+    // Find the last occurence of '.'
+    size_t last_pos = hostname.find_last_of(".");
+    if (last_pos != std::string::npos)
+    {
+        // Get a substring without last '.'
+        std::string substring = hostname.substr(0, last_pos);
+
+        // Find the last occurence of '.' in the substring created above.
+        size_t pos = substring.find_last_of(".");
+        if (pos != std::string::npos)
+        {
+            // Return the domain name between the last 2 occurences of '.' including the trailing dot'.'.
+            return std::string(hostname.substr(pos + 1, last_pos));
+        }
+    }
+    return std::string();
+}
+
 Global<MdnsContexts> MdnsContexts::sInstance;
 
 namespace {
 
+/**
+ * @brief Callback that is called when the timeout for resolving on the kOpenThreadDot domain has expired.
+ *
+ * @param[in] systemLayer The system layer.
+ * @param[in] callbackContext The context passed to the timer callback.
+ */
+void OpenThreadTimerExpiredCallback(System::Layer * systemLayer, void * callbackContext)
+{
+    ChipLogProgress(Discovery, "Mdns: Timer expired for resolve to complete on the open thread domain.");
+    auto sdCtx = static_cast<ResolveContext *>(callbackContext);
+    VerifyOrDie(sdCtx != nullptr);
+
+    if (sdCtx->hasOpenThreadTimerStarted)
```

Comment:
What does this check even mean?  If we got here, the timer was clearly started.....

Reply:
removed.